### PR TITLE
Add option to download and validate license from a remote location

### LINKED ISSUE #39: Download LICENSE.txt from a remote location
**Description:**
Currently, the update creator tool uses given license when creating updates. However, if the license gets updated users need to manually update the license when creating updates.This will get solved if the license is taken from the remote location when creating the updates.

**Suggested Labels:**
Priority/Normal
Severity/Minor
Type/Improvement

**Suggested Assignees:**
@mirage20

**Affected Product Version:** 1.0.1

### PR DESCRIPTION
**Pull request by Mirage20**
## Purpose
To get the latest valid license from a remote location so that developers do not need to find and add the latest license manually. Resolves #39

## Goals
To create updates with the valid latest license.

## Approach
Take the LICENSE.txt and MD5 checksum from a remote URL when creating and validating an update.

## User stories
- A user must be able to create an update without adding a valid license.
- A user must be able to validate license content.

## Release note
Add option to download and validate license from a remote location

## Documentation
Added to the README.md

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

## Samples
N/A

## Related PRs
https://github.com/wso2/update-creator-tool/pull/37

## Migrations (if applicable)
N/A

## Test environment
Ubuntu 16.04 x64
Go 1.8.3
 
## Learning
N/A

### CODE REVIEW

**Comment by Niranjan-K** [comment]
@miraj: What is the GO version that you used to test this?

File: README.md [hunk 1]
```diff
@@ -36,7 +36,11 @@ Optional Files -
 1. NOT_A_CONTRIBUTION.txt
 2. instructions.txt
 
-**Note:** You can generate the **update-descriptor.yaml** file using the **init** command as shown in the next section. You need to have the product(which you are creating the update for) distribution locally as well. This is used to compare files and create the proper file structure in the update zip.
+**Note 1:** You can generate the **update-descriptor.yaml** file using the **init** command as shown in the next section. You need to have the product(which you are creating the update for) distribution locally as well. This is used to compare files and create the proper file structure in the update zip.
+
+**Note 2:** If you do not have the correct license, the tool can download when creating the update. In order for the tool to download the license,  please set LICENSE_URL environment variable to the following URL.
+
+    https://staging-cdn-updates.private.wso2.com/wum/update/license/LICENSE.txt
```
Comment (Niranjan-K) [inline README.md]: This URL is only accessible from WSO2 network. Therefore if we need to do this offline, then what is the process?

File: README.md [hunk 2]
```diff
@@ -129,7 +133,10 @@ If the **UPDATE_LOCATION** contained the update 0001, by running this command, y
 
 After we create a update, we might want to unzip it and add more detail to the **update-descriptor.yaml** like removed files. After we do these changes, we can use this validation command to verify that the file structure of the zip is is the same as the distribution. 
 
-The `LICENSE_MD5` environment variable should be set to the expected MD5 license checksum in order to validate the LICENSE.txt contents.
+Please set `LICENSE_URL` environment variable to the following URL if you want to validate the LICENSE.txt contents.
+
+    https://staging-cdn-updates.private.wso2.com/wum/update/license/LICENSE.txt
```
Comment (Niranjan-K) [inline README.md]: Can we do the validation offline?

File: README.md (hunk 1 above)
Comment (Mirage20) [reply README.md]: The offline process is the same. Developers need to have some license file to create the update if they are offline.

File: README.md (hunk 2 above)
Comment (Mirage20) [reply README.md]: Offline validation only checks the existence of the file. Developers need to be in online in order to validate with the latest license.

**Comment by Mirage20** [comment]
Closing this PR until the design gets finalized.

**Pull request closed without merging**